### PR TITLE
Hide Transaction history when on testnet

### PR DESCRIPTION
**Pull request by osinakayah**
Transactions history should not be displayed when the wallet is switched to Testnet

### CODE REVIEW

File: src/features/accounts/AccountDetailsScreen.js [hunk 1]
```diff
@@ -143,11 +144,15 @@ export function filterTransactionHistory(transactions, accountAddress) {
 
 function TransactionHistory({accountAddress}) {
   const allTransactions = useSelector(transactionsSelectors.getTransactions);
+  const networkId = useSelector(appSelectors.getNetworkId);
   const transactions = useMemo(() => {
     return filterTransactionHistory(allTransactions, accountAddress);
   }, [allTransactions, accountAddress]);
 
   return useMemo(() => {
+    if (networkId !== 'mainnet') {
```
[the remark this answers is not in the document]
Comment (osinakayah) [reply src/features/accounts/AccountDetailsScreen.js:153]: Yes, I will. Let me confirm from @maycon-mello what he is proposing.
If we should create it in an env config or create a UI option in the Dev settings to toggle the  transaction history view.

[the remark this answers is not in the document]
Comment (mike-parkhill) [reply src/features/accounts/AccountDetailsScreen.js:153]: I think a Dev setting would be ideal so we can change it at runtime.

[the remark this answers is not in the document]
Comment (osinakayah) [reply src/features/accounts/AccountDetailsScreen.js]: @mike-parkhill 
I have included @maycon-mello  suggestion that we discussed on slack.

File: src/features/accounts/AccountDetailsScreen.js [hunk 2]
```diff
@@ -145,13 +145,20 @@ export function filterTransactionHistory(transactions, accountAddress) {
...
     return filterTransactionHistory(allTransactions, accountAddress);
   }, [allTransactions, accountAddress]);
 
   return useMemo(() => {
-    if (networkId !== 'mainnet') {
-      return <NBox />;
+    if (networkId !== 'mainnet' && !showTestnetConfig) {
+      return (
+        <NBox>
+          <Typography variant="list-description">Do not show</Typography>
```
Comment (mike-parkhill) [inline src/features/accounts/AccountDetailsScreen.js:159]: text should be in a translations file and the message should probably be something like, "History is hidden on <network_id>"

File: src/features/dev-settings/DevSettingsScreen.js [hunk 3]
```diff
@@ -140,6 +166,46 @@ export function DevSettingsScreen({onAddAccount, onNetworkChange}) {
           </Stack>
         ) : null}
 
+        {showTransactionHistory ? (
```
Comment (mike-parkhill) [inline src/features/dev-settings/DevSettingsScreen.js:169]: should we hide the option if the user is on mainnet?

Comment (osinakayah) [reply src/features/dev-settings/DevSettingsScreen.js:169]: That actually sounds better. Got it.

Comment (osinakayah) [reply src/features/dev-settings/DevSettingsScreen.js:173]: > should we hide the option if the user is on mainnet?

This is has been implemented now

**Review by mike-parkhill — APPROVED** [review]

**Review by mike-parkhill — APPROVED** [review]

**Pull request merged**